views/js/runner/plugins/tools/textToSpeech/textToSpeech.js

```diff
@@ -0,0 +1,374 @@
+            },
+
+            /**
+             * Play
+             */
+            play: function play() {
+                if (!this._exec('isSpeaking') || this._exec('isPaused')) {
+                    this._exec('event_play');
+                    this.trigger('play');
+                }
+            },
+
+            /**
+             * Pause
+             */
+            pause: function pause() {
+                this._exec('event_pause');
+                this.trigger('pause');
+            },
+
+            /**
+             * Stop
+             */
+            stop: function stop() {
+                this._exec('stopSpeech');
+                this.trigger('stop');
+            },
+
+            /**
+             * Speech speed
+             */
+            setSpeed: function setSpeed(e, value) {
+                speed = +value;
+
+                this._exec('setSpeedValue', speed);
+                this.trigger('setSpeed', speed);
+            },
+
+            /**
+             * Volume
+             */
+            setVolume: function setVolume(e, value) {
+                volume = +value;
+
+                this._exec('setVolumeValue', volume);
+                this.trigger('setVolume', volume);
+            },
+
+            /**
+             * Click to pronounce
+             */
+            clickToSpeak: function clickToSpeak() {
+                options.enableClickToSpeak = !options.enableClickToSpeak;
+                this._exec('enableClickToSpeak', options.enableClickToSpeak);
+                this._exec('enableContinuousReading', !options.enableClickToSpeak); // continuous reading is off when click to speak is on
+                this.trigger('clickToSpeak');
+            }
+        }, {
+            // defaults
+        })
+        .setTemplate(tpl)
+        .on('init', function () {
+            var self = this;
+
+            speed = 40; // default speed
+            volume = 40; // default volume
+
+            // we have to mark some blocks as ignored to prevent TTS accessing it
+            $(options.ignoreEls.join(',')).each(function () {
+                $(this).attr('ignore', true);
+            });
+
+            require([options.toolbarUrl], function () {
+                var tss = self._get('texthelpSpeechStream');
+
+                require(["//" + tss.g_strServer + "/SpeechStream/v" + tss.g_strBuild + "/texthelpMain.js"], function () {
+                    self._init();
+                });
+            });
+        })
+        .init(config)
+        .on('render', function () {
+            var $this = this.getElement();
+
+            // prevents clicks from removing highlighted text
+            $this.on('mousedown', function (e) {
+                e.preventDefault();
+                return false;
+            });
+
+            // Action clicks
+            $this.find('.action').on('click', function (e) {
+                // prevents disabled actions from being triggered
```

Comment:
double instanciation of jquery container, it is preferable to have it build only once: 
```
var $action = $(this);
$action.doStuff();
$action.doOtherStuff();
```